### ISSUE TITLE
Es-dsl wildcards: Backslash and Asterisk

### ISSUE THREAD

**Issue by qux-bbb** [Neo23x0/sigma#387]
My raw sigma rule(test.yml) is:  
```yml
title: Test
status: experimental
description: Test Backslash and Asterisk
author: Test
logsource:
    product: windows
    service: sysmon
detection:
    selection:
        EventID: 13
        TargetObject: 
            - '*\Desktop\hello\*word'
    condition: selection
level: high
```
Use this to convert it to dsl file(test.json):
```sh
py -3 sigmac --target es-dsl -c elk-winlogbeat test.yml -o test.json
```
The result is:
```json
{
  "query": {
    "constant_score": {
      "filter": {
        "bool": {
          "must": [
            {
              "match_phrase": {
                "log_name": "Microsoft-Windows-Sysmon/Operational"
              }
            },
            {
              "match_phrase": {
                "event_id": "13"
              }
            },
            {
              "bool": {
                "should": [
                  {
                    "wildcard": {
                      "event_data.TargetObject.keyword": "*\\Desktop\\hello\\*word"
                    }
                  }
                ]
              }
            }
          ]
        }
      }
    }
  }
}
```
When I search by this json, it's not working.

**Comment by qux-bbb** [comment]
If I change yml to this:  
```yml
title: Test
status: experimental
description: Test Backslash and Asterisk
author: Test
logsource:
    product: windows
    service: sysmon
detection:
    selection:
        EventID: 13
        TargetObject: 
            - '*\\Desktop\\hello\\\*word'
    condition: selection
level: high
```
It will turn to:  
```json
{
  "query": {
    "constant_score": {
      "filter": {
        "bool": {
          "must": [
            {
              "match_phrase": {
                "log_name": "Microsoft-Windows-Sysmon/Operational"
              }
            },
            {
              "match_phrase": {
                "event_id": "13"
              }
            },
            {
              "bool": {
                "should": [
                  {
                    "wildcard": {
                      "event_data.TargetObject.keyword": "*\\\\Desktop\\\\hello\\\\\\*word"
                    }
                  }
                ]
              }
            }
          ]
        }
      }
    }
  }
}
```
And it works.

**Comment by thomaspatzke** [comment]
Some information about what should match would be helpful.

To me it appears as expected behavior:

* The first matches on a backslash followed by something and _word_
* The second matches on a backslash, followed by an asterisk (it's escaped) and _word_

Can you describe the difference to the expected result?

**Comment by qux-bbb** [comment]
What I want to match is `C:\User\anyuser1\Desktop\hello\anythingworld`

**Comment by thomaspatzke** [comment]
ES DSL appears to behave a bit unexpected here. I think I have to dig deeper into this when some things are finished or maybe @milkmix can help out? :wink:

**Comment by duttad** [comment]
I am also a little confused with the es-dsl wildcard issue. So, I am running the rule:  powershell_xor_commandline.yml that has the following lines:
```python
detection:
    selection:
        CommandLine:
            - '* -bxor*'
    condition: selection
```
and sigmac generates a es-dsl query where the wildcard part reads like below.
```python
"wildcard": {
    "CommandLine.keyword": "* -bxor*"
}
```
I was expecting the generated query to read something like:
```python
"wildcard": {
            "CommandLine": {
                "value": "* -bxor*",
                "boost": 1.0,
                "rewrite": "constant_score"
            }
}
```
based on the Elasticsearch 7.2 documentation: [https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-wildcard-query.html](Elasticsearch Reference [7.2] » Query DSL » Term-level queries » Wildcard Query)

**Comment by michielmeersmans** [comment]
It appears you need to escape a backslash within a wildcard query. Since the DSL is json, those backslashes also need to be escaped. So for your example, the query would become:
```json
{
    "wildcard": {
      "event_data.TargetObject.keyword": "*\\\\Desktop\\\\hello\\\\*word"
    }
 }
```

**Comment by qux-bbb** [comment]
Yeah. But it doesn't work. If I write like this, it works.  
```json
{
    "wildcard": {
        "event_data.TargetObject.keyword": "*\\\\Desktop\\\\hello\\\\\\*word"
    }
}
```

**Comment by thomaspatzke** [comment]
Fixed by #416
